Assets/00_Content/Scripts/Vikings/States/BrawlingVikingState.cs

```diff
@@ -64,6 +100,23 @@ public class BrawlingVikingState : VikingState {
 			return this;
 		}
 
+		private VikingState DoVikingBrawl() {
+			Debug.Log("Fighting another viking...");
+			return this;
+		}
+
+		private VikingState DoPlayerBrawl() {
+
+			if ((navMeshAgent.transform.position - playerTarget.transform.position).sqrMagnitude < viking.Data.attackTriggerDistance && !viking.IsAttacking) {
+				viking.MakeSpinAttack();
+				return this;
+			}
+
+			navMeshAgent.SetDestination(playerTarget.transform.position);
```

Comment:
Recalculating the path every frame could probably be very expensive. See how I handled this in `BeingSeatedVikingState.Update()`.

Reply:
yes good catch!